course/page/upload.py

```diff
@@ -294,7 +310,7 @@ def normalized_bytes_answer(self, page_context, page_data, answer_data):
 
         subm_data, subm_mime = self.get_content_from_answer_data(answer_data)
 
-        from mimetypes import guess_extension
+        add_type("application/x-ipynb+json", ".ipynb")
```

Comment:
Could you move this into the module root?

Reply:
The import is already at the module level.  Are you also wanting `add_type()` called at the module root?

Reply:
Yes, so that there's not a behavior change for the other uses of `mimetypes` based on whether this method has been called.